### DISCUSSION TITLE
Limit to number of menu items?

### DISCUSSION THREAD

**Discussion by ckiernan** [VasilKalchev/LiquidMenu#83] [Q&A]
Hi , I've developed a menu using LiquidLine and LiquidScreen for a 16x2 lcd based on your D_button example that has 15 menu definitions since it appeared to be the easiest way to enter variables. The menu works fine displaying the variable name on line 1 and the value on line 2 but will not show menu items past the 13th menu item. I'm attaching the code since its broken up into header files for integration into another code base. 
Is this a limitation in the number of menu items I'm hitting up against or am I not doing something correctly. I'm trying to develop a nested menu using the H_System menu example J_Scrolling menu examples but that's getting complicated for me. Any help would be appreciated.

[LiquidMenu_V1.2.zip](https://github.com/VaSe7u/LiquidMenu/files/7909112/LiquidMenu_V1.2.zip)

**Comment by VasilKalchev** [comment]
Hello, @ckiernan and sorry for the late reply.

The library uses an array of fixed size to hold pointers to the screens. The size of the array is specified in "src/LiquidMenu_config.h" with `const uint8_t MAX_SCREENS = 14;` and is set to `14` by default. Increasing the number should fix your problem at the expense of higher memory usage. You can adjust the values of the other `MAX_...` variables in the same file to conserve memory.

If this doesn't help, let me know.
